StRoot/StSpinPool/StFcsPi0FinderForEcal/StFcsPi0FinderForEcal.cxx

```diff
@@ -0,0 +1,507 @@
+//class StFcsPi0FinderForEcal
+//author Xilin Liang
+//
+//critical plots for offline QA: h1_inv_mass_cluster , h1_two_cluster_energy_allcut , h1_dgg_cluster  , h2_cluster_dgg_vs_E1pE2
+//
+//
+
+#include "StFcsPi0FinderForEcal.h"
+
+#include "StEvent/StEnumerations.h"
+#include "StEvent/StEvent.h"
+#include "StEvent/StFcsCluster.h"
+#include "StEvent/StFcsCollection.h"
+#include "StEvent/StFcsHit.h"
+#include "StEventTypes.h"
+#include "StFcsDbMaker/StFcsDbMaker.h"
+#include "StMessMgr.h"
+#include "StMuDSTMaker/COMMON/StMuTypes.hh"
+#include "StSpinPool/StFcsQaMaker/StFcsQaMaker.h"
+#include "StSpinPool/StFcsRawDaqReader/StFcsRawDaqReader.h"
+#include "StThreeVectorF.hh"
+#include "Stypes.h"
+#include "TBox.h"
+#include "TCanvas.h"
+#include "TColor.h"
+#include "TFile.h"
+#include "TH1.h"
+#include "TH2.h"
+#include "TLine.h"
+#include "TMarker.h"
+#include "TROOT.h"
+#include "TString.h"
+#include "TStyle.h"
+#include "TText.h"
+
+#ifndef SKIPDefImp
+ClassImp(StFcsPi0FinderForEcal)
+#endif
+
+    //------------------------
+    StFcsPi0FinderForEcal::StFcsPi0FinderForEcal(const Char_t* name) : StMaker(name) {
+}
+
+StFcsPi0FinderForEcal::~StFcsPi0FinderForEcal() {}
+
+//-----------------------
+Int_t StFcsPi0FinderForEcal::Init() {
+   mFcsDb = static_cast<StFcsDb*>(GetDataSet("fcsDb"));
+   mFcsDb->setDbAccess(0);
+   if (!mFcsDb) {
+      LOG_ERROR << "StFcsEventDisplay::InitRun Failed to get StFcsDbMaker" << endm;
+      return kStFatal;
+   }
```

Comment:
Makes no sense to first dereference a pointer and then check it for NULL.

Reply:
Agreed